pkg/sqlutil/sqlutil.go

```diff
@@ -0,0 +1,61 @@
+package sqlutil
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	"github.com/jmoiron/sqlx"
+)
+
+// Queryer is implemented by [*sqlx.DB], [*sqlx.Tx], & [*sqlx.Conn].
+type Queryer interface {
+	sqlx.ExtContext
+	sqlx.PreparerContext
+	GetContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error
+	SelectContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error
+}
+
+type TxOptions struct {
+	sql.TxOptions
+	OnPanic func(recovered any, rollbackErr error)
```

Comment:
This hook allows core to reproduce the existing behavior for sonar reporting and logging.